src/libgit2/transports/ssh_libssh2.c

```diff
@@ -539,6 +541,10 @@ static int _git_ssh_session_create(
 		return -1;
 	}
 
+	if (git_socket_stream__connect_timeout > 0) {
+		libssh2_session_set_timeout(s, git_socket_stream__connect_timeout);
```

Comment:
We should inherit the `git_socket_stream__connect_timeout` from the socket stream. So we already have connection timeouts.

I think that for this, we want to use `GIT_OPT_SET_SERVER_TIMEOUT`. So I think that this should be `git_socket_stream__timeout` instead...

Reply:
updated PR